overlord/devicestate/handlers_install.go

```diff
@@ -435,6 +469,9 @@ func (m *DeviceManager) doRestartSystemToRunMode(t *state.Task, _ *tomb.Tomb) er
 	if err := writeLogs(boot.InstallHostWritableDir); err != nil {
 		logger.Noticef("cannot write installation log: %v", err)
 	}
+	if err := writeTimings(boot.InstallHostWritableDir, st); err != nil {
```

Comment:
given that we can't hold the lock this probably needs to be moved at the end of the whole handler and we should set the status to done and then release the lock before calling it

Reply:
this is a bit complicated because we close the sockets when we restart, so this probably needs to be done before requesting the restart, so the lock handling needs to be a bit more involved

Reply:
Yeah, we need to do it before we ask snapd to restart, I had this idea to use a defer but that won't work :/